core/transaction/transaction.go

```diff
@@ -1,11 +1,11 @@
 package transaction
 
 import (
-	"github.com/cosmos/gogoproto/proto"
+	gogoproto "github.com/cosmos/gogoproto/proto"
```

Comment:
@tac0turtle is it possible to decouple

cosmossdk.io/core and gogoproto dependency tree wise?

Reply:
Gogoproto is already a type within core. Since it's directly related to how the system works it's not needed to decouple. This import also does not prevent compiling down to rv32. What is the reason you would want to avoid this?

Reply:
Don't see why it can't be 

type Marshallalable interface {
Marshal() 
Unmarshal()
} 

Having core abstracted from any implementation details around serialization seems like good practice. 

cosmossdk.io/core should be over the wire agnostic imo.

Reply:
If we continue on the path of enshrining protobuf into base types of the system we will never get rid of the nightmares.

Reply:
proto.message is an interface, 

```go
type Message interface {
	Reset()
	String() string
	ProtoMessage()
}
```

its annoying I understand but this isn't defining serialisation either. 

Marshal and unmarshal dont work here since you cant assume a type has those